src/inc/til/u8u16convert.h

```diff
@@ -23,243 +23,28 @@ Author(s):
 
 namespace til // Terminal Implementation Library. Also: "Today I Learned"
 {
-    template<class charT>
-    class u8u16state final
+    // state structure for maintenance of UTF-8 partials
+    struct u8state
     {
-    public:
-        u8u16state() noexcept :
-            _buffer{},
-            _utfPartials{}
-        {
-        }
+        std::array<char, 4> partials{}; // first 3 bytes are for the code units of a partial code point, last byte is the number of cached code units

```

Comment:
@miniksa made this a std::array after a static analysis. However, we don't use any features of the std::array, so we can IMO turn this into a simple array.

Reply:
Oh that's OK. If you like the simple array better, go for it.